python/az/aro/azext_aro/_validators.py

```diff
@@ -283,3 +289,74 @@ def validate_load_balancer_managed_outbound_ip_count(namespace):
     if namespace.load_balancer_managed_outbound_ip_count < minimum_managed_outbound_ips or namespace.load_balancer_managed_outbound_ip_count > maximum_managed_outbound_ips:  # pylint: disable=line-too-long
         error_msg = f"--load-balancer-managed-outbound-ip-count must be between {minimum_managed_outbound_ips} and {maximum_managed_outbound_ips} (inclusive)."  # pylint: disable=line-too-long
         raise InvalidArgumentValueError(error_msg)
+
+
+def validate_enable_managed_identity(namespace):
+    if not namespace.enable_managed_identity:
+        return
+
+    if namespace.client_id is not None:
+        raise InvalidArgumentValueError('Must not specify --client-id when --enable-managed-identity is True')
+
+    if namespace.client_secret is not None:
+        raise InvalidArgumentValueError('Must not specify --client-secret when --enable-managed-identity is True')
+
+    if namespace.version is None or not re.match(r'^[4-9]{1}\.[0-9]{1,2}\.[0-9]{1,2}$', namespace.version):
+        raise InvalidArgumentValueError('Enabling managed identity requires --version >= 4.14.z')
+
+    _, versionY, _ = namespace.version.split('.', 2)
+    if int(versionY) < 14:
+        raise InvalidArgumentValueError('Enabling managed identity requires --version >= 4.14.z')
+
+    if not namespace.platform_workload_identities:
+        raise RequiredArgumentMissingError('Enabling managed identity requires platform workload identities to be provided')  # pylint: disable=line-too-long
+
+    if not namespace.mi_user_assigned:
+        raise RequiredArgumentMissingError('Enabling managed identity requires cluster identity to be provided')
+
+
+def validate_platform_workload_identities(cmd, namespace):
+    if namespace.platform_workload_identities is None:
+        return
+
+    if not namespace.enable_managed_identity:
+        raise RequiredArgumentMissingError('Must set --enable-managed-identity when providing platform workload identities')  # pylint: disable=line-too-long
+
+    for identity in namespace.platform_workload_identities:
+        if not is_valid_resource_id(identity.resource_id):
+            identity.resource_id = identity_name_to_resource_id(
```

Comment:
It seems it does more than validation.
I think it would be better that we stick to validation in validate functions, and if we need formalization, do somewhere else, otherwise it will be confusing.

btw is it safe to assume that `identity.resource_id` is a MI name when it is not valid?

Reply:
I did this to keep in line with our "validation" performed on vnets/subnets (that does the same name -> resourceid conversion): https://github.com/Azure/ARO-RP/blob/tsatam/ARO-4398-wimi-cli-p1-base-api-functionality/python/az/aro/azext_aro/_validators.py#L239

I agree that this is confusing, it tripped me up when implementing this. Perhaps we move these conversions to the Actions implementation used above? 

This is something I'd like to tackle in an overall refactor, but perhaps not within this PR itself.

Reply:
As for our name / resource ID handling here, we are not yet validating that the identity resource even exists yet. 

Ultimately this will be handled by API-side validation, and we will introduce checks for the identity existing further along in MIWI efforts, once we introduce validation of the passed-in identities and their roleassignments.

Reply:
thanks! fair enough

> Perhaps we move these conversions to the Actions implementation used above?
It would be better I guess.

I was worried about `identity.resource_id` conversion because if there's just a typo, it could be interpreted as a resource name.
However, on second thoughts, it doesn't create anything, and even though there's a typo, it'll end up just getting an error, so it isn't a big deal.